exporter/datadogexporter/internal/translator/metrics_translator.go

```diff
@@ -48,12 +44,11 @@ type Translator struct {
 	prevPts                  *TTLCache
 	logger                   *zap.Logger
 	cfg                      config.MetricsConfig
-	buildInfo                component.BuildInfo
 	fallbackHostnameProvider HostnameProvider
 }
```

Comment:
(discussed offline but adding comment here for visibility) This would need a major refactor on how the exporter works. If we went that route, since multiple goroutines may be calling `MapMetrics` at the same time: 
1. the `Consumer` would need  to be thread-safe,
2. it would need to have some sort of batching to accumulate and send metrics,
3. we would need to handle `running` metrics differently.

Passing the `Consumer` explicitly allows us to avoid doing this. In the Agent we will use the same `Consumer` everytime.